assets/components/button/dangerouslySetButtonOnAnyElement.jsx

```diff
@@ -14,48 +14,45 @@ import './button.scss';
 type Appearance = 'primary' | 'green' | 'greenHollow' | 'greyHollow';
 type IconSide = 'left' | 'right';
 
-type PropTypes = {|
+type GenericPropTypes = {|
   children: Node,
   icon?: Node,
   appearance: Appearance,
   iconSide: IconSide,
   modifierClasses: string[],
 |};
 
-type DangerousPropTypes = {
+type PropTypes = {
   element: 'a' | 'button' | 'div',
```

Comment:
Why can't element be part of GenericPropTypes? Then you could remove the need for a separate `PropTypes` and `GenericPropTypes`

Reply:
`GenericPropTypes` is exposed as the proptypes for all buttons, and we don't want consumers of, say `linkButton` specifying an element